File: TheLionsDen.Model/Responses/RoomTypeResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TheLionsDen.Model.Responses
{
    public class RoomTypeResponse
    {
        public int RoomTypeId { get; set; }
        public string Name { get; set; } 
        public int Capacity { get; set; }
        public int Size { get; set; }
        public string Description { get; set; }
        public string Rules { get; set; }

        public string SizeM2 => $"{Size}m2";

        public virtual ICollection<RoomImageResponse> RoomImages { get; set; }
        //public virtual ICollection<Room> Rooms { get; set; }
    }
}

File: the_lions_den_mobile/lib/model/room_type/room_type_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:the_lions_den_mobile/model/room_image/room_image_response.dart';
part 'room_type_response.g.dart';

@JsonSerializable()
class RoomTypeResponse {
  int? roomTypeId;
  String? name;
  String? description;
  String? rules;
  int? size;
  int? capacity;

  List<RoomImageResponse>? roomImages;

  RoomTypeResponse();

  factory RoomTypeResponse.fromJson(Map<String, dynamic> json) =>
      _$RoomTypeResponseFromJson(json);

  /// Connect the generated [_$RoomTypeResponseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RoomTypeResponseToJson(this);
}
